IRCT-CL/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/cl/util/Utilities.java

```diff
@@ -259,4 +273,44 @@ public static String extractUserFromTokenIntrospection(HttpServletRequest req, S
 
 		return null;
 	}
+
+	/**
+	 * to apply a proxy to a http function by reading the environment variables from JVM
+	 * @param request
+	 */
+	public static void applyProxySettings(HttpRequestBase request) {
+		String proxyHost = System.getProperty("http.proxyHost");
+		String proxyPort = System.getProperty("http.proxyPort");
+		String proxyProtocol = System.getProperty("http.proxyProtocol"); // non-standard
+		if (proxyHost != null) {
+			int port = 80;
+			if (proxyPort != null) {
+				port = Integer.valueOf(proxyPort);
```

Comment:
as `port` is of primitive int type. You should ideally use `Integer.parseInt` instead of `Integer.valueOf` which will return Integer.